### PR TITLE
Added telemetry functionality for mission control

### PR DESCRIPTION
**Pull request by vvarun77**
Added `teleSlice.js` and `telemetryMiddleware.js` to mission control for storing and receiving telemetry data relating to the rover's orientation and position.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,6 +194,32 @@ Sent from the rover server to inform Mission Control of a motor's status.
...
+}
+```
+### Parameters
+- `orientW` - refers to the orientation quaternion W component
+- `orientX` - refers to the orientation quaternion X component
+- `orientY` - refers to the orientation quaternion Y component
+- `orientZ` - refers to the orientation quaternion Z component
+- `posX` - refers to the X position of the rover in world reference frame
+- `posY` - refers to the Y position of the rover in world reference frame
+- `posZ` - refers to the Z position of the rover in world reference frame
```
Comment (abhaybd) [inline README.md:221]: I think you forgot to add the time field to the documentation as well

File: src/store/middleware/telemetryMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        store.dispatch(roverPositionReportReceived({
+          poseReport: {
+            orientW,
+            orientX,
+            orientY,
+            orientZ,
+            posX,
+            posY,
+            posZ,
+            timestamp
```
Comment (abhaybd) [inline src/store/middleware/telemetryMiddleware.js:20]: Can we name this something else? Like we discussed, this isn't going to be timestamp, it'll be something more like data age or recency.

File: src/store/teleSlice.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    orientX: null,
+    orientY: null,
+    orientZ: null,
+    posX: null,
+    posY: null,
+    posZ: null,
+    timestamp: null
+}
+
+const teleSlice = createSlice({
```
Comment (abhaybd) [inline src/store/teleSlice.js:14]: `tele` is an awkward prefix. Can we call this `telemetrySlice`? It's not *that* long and it's more explicit.

**Review by abhaybd — CHANGES_REQUESTED** [review]
Overall pretty good! Had some minor issues regarding documentation and some naming. Also, can we rename `teleSlice.js` to `telemetrySlice.js`? It's more readable and explicit.

File: README.md [hunk 4]
```diff
@@ -219,6 +220,7 @@ Sent from the rover to inform Mission Control of the rover's current position in
 - `posX` - refers to the X position of the rover in world reference frame
 - `posY` - refers to the Y position of the rover in world reference frame
 - `posZ` - refers to the Z position of the rover in world reference frame 
+- `recency` - refers to the difference in time between when the message was sent and received
```
Comment (abhaybd) [inline README.md:223]: This description isn't quite right, that would be something closer to latency. Recency measures the amount of elapsed time between the value being measured and being sent from the rover.

**Review by abhaybd — CHANGES_REQUESTED** [review]
Looks better, just a minor change needed about the documentation.

I'm not too familiar with redux, so I couldn't entirely say if the changes to the `store.js` are needed, so I'll defer that to Kyle. I'd say fix the documentation issue and get a final signoff from Kyle and then merge it.

File: README.md [hunk 5]
```diff
@@ -220,7 +220,7 @@ Sent from the rover to inform Mission Control of the rover's current position in
 - `posX` - refers to the X position of the rover in world reference frame
 - `posY` - refers to the Y position of the rover in world reference frame
 - `posZ` - refers to the Z position of the rover in world reference frame 
-- `recency` - refers to the difference in time between when the message was sent and received
+- `recency` - refers to the difference in time between when the measurement was taken and sent
```
Comment (abhaybd) [inline README.md:223]: Should have mentioned this before, but we need units. For `posX`, `posY`, and `pozZ` can we specify they're in meters, and `recency` is in seconds?

**Review by abhaybd — CHANGES_REQUESTED** [review]
One minor thing I forgot!

**Review by KyleR56 — APPROVED** [review]
Looks good aside from some small changes which I just made myself in a new commit (I also fixed some small style issues I found in other files). There might be a slight naming inconsistency with "telemetry" and "roverPosition". The two names are used interchangeably, so we might want to just use 1 of them for consistency. But I don't think it's a big deal so I'll leave it up in the air.

**Review by abhaybd — APPROVED** [review]
Looks good!

**Pull request merged**